### ISSUE TITLE
"--softmax --lambda -1.0" breaks CLI

### ISSUE THREAD

**Issue by marco-rossi29** [VowpalWabbit/vowpal_wabbit#1860]
when using `--cb_explore_adf --softmax` the default `--lambda` is `-1.f`.

However VW fails when `--lambda -1.0` is in input string

```
C:\work\vw\vowpalwabbit\out\target\Release\x64\vw.exe --cb_adf --lambda -1.0
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile =
num sources = 1
Error: unrecognised option '---1'

finished run
number of examples = 0
weighted example sum = 0.000000
weighted label sum = 0.000000
average loss = n.a.
total feature number = 0
unrecognised option '---1'
vw (options_boost_po.cc:152): unrecognised option '---1'
```

VW doesn't recognize that `-1.0` is the argument of `--lambda`

I believe the issue is in parsing `-1.0` versus `-1`, since`--cb_explore_adf --softmax --lambda -1` works

**Comment by jackgerrits** [comment]
I'll look into this

**Comment by lokitoth** [comment]
@jackgerrits Take a look at #1742: I thought I had fixed this, but it looks like it was not a complete fix?

**Comment by JohnLangford** [comment]
You can pass a negative number by using --lambda=-1.0 .

Another alternative, btw, is that we could add a negative sign into the definition of softmax so the correct range of values is positive.

**Comment by JohnLangford** [comment]
I'm implementing the alternative---it seems like the better long term solution.
